egs/wsj/s5/steps/conf/train_calibration.sh

```diff
@@ -0,0 +1,120 @@
+#!/bin/bash
+# Copyright 2015, Brno University of Technology (Author: Karel Vesely). Apache 2.0.
+
+# Trains logistic regression, which calibrates the per-word confidences in 'CTM'.
+# The 'raw' confidences are obtained by Minimum Bayes Risk decoding.
+
+# The input features of logistic regression are:
+# - logit of Minumum Bayer Risk posterior
+# - log of word-length in characters
+# - log of average-depth depth of a lattice at words' position
+# - log of frames per character ratio
+# (- categorical distribution of 'lang/words.txt', DISABLED)
+
+# begin configuration section.
+cmd=
+lmwt=12
+decode_mbr=true
+word_min_count=10 # Minimum word-count for single-word category,
+normalizer=0.0025 # L2 regularization constant,
+stage=0
+# end configuration section.
+
+[ -f ./path.sh ] && . ./path.sh
+. parse_options.sh || exit 1;
+
+if [ $# -ne 7 ]; then
+  echo "Usage: $0 [opts] <data-dir> <lang-dir|graph-dir> <word-filter> <word-length> <unigrams> <decode-dir> <calibration-dir>"
```

Comment:
some documentation of the format of the inputs would be nice, since some of them are non-standard.
However, I think it would be possible to make the input format more general and also easier to work with.
Couldn't the input format be just a file containing:
integer-word-id  feature1 feature2 ... featureN?
That way you wouldn't be filtering out words like <cough> or partial words, you'd just give them their own feature (which presumably would have a negative coefficient), and that would produce a better model for cases where you ended up wanting to predict them.
(BTW, the program 'paste' should be useful here).